### DISCUSSION TITLE
Worktree HEAD traceability

### DISCUSSION THREAD

**Discussion by miquella** [arxanas/git-branchless#1108] [Ideas]
Recently I was thinking about #928 and realized something that may be missing.

When running `smartlog`, by default it shows anything traceable from my current HEAD as well as any local branches, correct?

One of the core reasons to use worktrees is so that all of the commits from one worktree are immediately available in other worktrees from the same repo. So it strikes me that perhaps `smartlog` should also show anything traceable from any worktree's HEAD.

Each worktree's HEAD could be denoted similar to branches, that way when switching between worktrees, you don't lose track of recently committed work (I employ _a lot_ of worktrees, so I feel this pain often 😆).

**Comment by arxanas** [comment]
This sounds reasonable, although there is currently no query to find all worktree `HEAD`s. I would accept a PR implementing this. We would probably have to define a revset that expands to all worktree `HEAD`s along the way.

In theory, with https://github.com/arxanas/git-branchless/pull/1095, we would reflect new commits in any worktree across all worktrees, so you would be able to see all work in progress from any worktree (although this is a different thing from showing all worktree `HEAD`s; for example, it wouldn't show branches that were pulled from a remote and not registered via the `post-commit` hook).
